Source/JavaScriptCore/dfg/DFGSpeculativeJIT32_64.cpp

```diff
@@ -3824,28 +3825,66 @@ void SpeculativeJIT::compile(Node* node)
         flushRegisters();
         GPRFlushedCallResult result(this);
         GPRReg resultGPR = result.gpr();
-        if (node->child1().useKind() == MapObjectUse)
-            callOperation(operationJSMapFindBucket, resultGPR, LinkableConstant::globalObject(*this, node), mapGPR, keyRegs, hashGPR);
-        else
-            callOperation(operationJSSetFindBucket, resultGPR, LinkableConstant::globalObject(*this, node), mapGPR, keyRegs, hashGPR);
-        cellResult(resultGPR, node);
+        auto operation = isMapObjectUse ? operationMapKeyIndex : operationSetKeyIndex;
+        callOperation(operation, resultGPR, LinkableConstant::globalObject(*this, node), mapGPR, keyRegs, hashGPR);
+        exceptionCheck();
+        strictInt32Result(resultGPR, node);
```

Comment:
Let's change it to new Operation form which does exception check with returned value.